apps/web/src/app/[locale]/signups/[signupId]/[signupEditToken]/signup-form.tsx

```diff
@@ -0,0 +1,295 @@
+                required
+                defaultValue={signup.firstName ?? undefined}
+              />
+              {state?.errors?.firstName ? (
+                <span aria-live="polite" className="block text-red-600">
+                  {state.errors.firstName.join(", ")}
+                </span>
+              ) : null}
+            </p>
+            <p className="w-full max-w-sm space-y-2">
+              <label className="block" htmlFor="lastName">
+                {t("Last name")}
+              </label>
+              <Input
+                id="lastName"
+                name="lastName"
+                type="text"
+                placeholder="Teekkari"
+                autoComplete="family-name"
+                required
+                defaultValue={signup.lastName ?? undefined}
+              />
+              {state?.errors?.lastName ? (
+                <span aria-live="polite" className="block text-red-600">
+                  {state.errors.lastName.join(", ")}
+                </span>
+              ) : null}
+            </p>
+            <p className="w-full max-w-sm space-x-2">
+              <Checkbox
+                id="namePublic"
+                name="namePublic"
+                defaultChecked={signup.namePublic}
+              />
+              <label htmlFor="namePublic">
+                {t("Show name in the public list of sign ups")}
+              </label>
+            </p>
+          </>
+        )}
+        {!!event.emailQuestion && (
+          <p className="w-full max-w-sm space-y-2">
+            <label className="block" htmlFor="email">
+              {t("Email")}
+            </label>
+            <Input
+              id="email"
+              name="email"
+              type="email"
+              autoComplete="email"
+              placeholder="teemu.teekkari@aalto.fi"
+              required
+              defaultValue={signup.email ?? undefined}
+            />
+            {state?.errors?.email ? (
+              <span aria-live="polite" className="block text-red-600">
+                {state.errors.email.join(", ")}
+              </span>
+            ) : null}
+          </p>
+        )}
+        {event.questions.map((question) => (
+          <InputRow
+            key={question.id}
+            question={question}
+            defaultValue={
+              signup.answers.find((answer) => answer.questionId === question.id)
+                ?.answer ?? undefined
+            }
+            errors={state?.errors?.[question.id]}
+          />
+        ))}
+        <p className="w-full max-w-sm">
+          {t(
+            "You can edit your sign up or delete it later from this page, which will be sent to your email in the confirmation message",
+          )}
+        </p>
+        <SubmitButton />
+        <Button
+          formNoValidate
+          // eslint-disable-next-line @typescript-eslint/no-misused-promises -- server actions can be ignored promises
+          formAction={deleteAction}
```

Comment:
```suggestion
          formAction={void deleteAction}
```

Reply:
Also I wonder if we could add the same "are you sure" functionality here as there is with the original ilmomasiina form? To prevent accidental cancellations.

Reply:
The double button click behavior of ilmomasiina is a bit odd, we had a design for a confirmation modal/popover by @ellimattila that we could implement here

Reply:
I don't think `void deleteActioṉ` works here, but I could be wrong. I recall last time I tried it, it broke the server action completely